pages/product/edit/[id].js

```diff
@@ -0,0 +1,45 @@
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import ProductForm from "@/components/ProductForm";
+import Link from "next/link";
+
+export default function EditProduct() {
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: product, isLoading, error } = useSWR(`/api/shoppinglist/${id}`);
+
+  if (error) {
+    return <p>We couldn`t edit the item...</p>;
```

Comment:
here the error might have multiple causes, not just an error in editing. please rephrase more generic like "An error occurred"